### PR TITLE
Upgrade to GWT 2.8

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?

- Upgrade to `GWT` 2.8
- `GWT ClosureCompiler` was removed
- `maven.compiler.source` and `maven.compiler.target` were upgraded to 1.8
- Added new dependency `com.google.code.findbugs:jsr305:3.0.1` for supporting `Nonnull`, `Nullable` annotations and Guava Preconditions

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/2869

#### Changelog
Upgraded GWT to version 2.8 including new `guava` and `lib-gwt-svg`.

#### Release Notes

n/a

#### Docs
n/a

### CODE REVIEW

**Comment by skabashnyuk** [comment]
move gwt.compiler.version to separate property

**Comment by skabashnyuk** [comment]
add 
```               
 <groupId>com.google.code.findbugs</groupId>
 <artifactId>jsr305</artifactId> 

```
to war packaging exclude

File: assembly/assembly-ide-war/pom.xml [hunk 1]
```diff
@@ -23,8 +23,8 @@
     <name>Che IDE :: Compiling GWT Application</name>
     <properties>
         <generated.sources.directory>${project.build.directory}/generated-sources/gen</generated.sources.directory>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline assembly/assembly-ide-war/pom.xml]: variable

File: assembly/assembly-ide-war/pom.xml [hunk 2]
```diff
@@ -434,7 +434,6 @@
                     <!-- don' remove it we will use it then need to found bug in compiled JS -->
                     <!--style>DETAILED</style-->
                     <logLevel>${gwt.compiler.logLevel}</logLevel>
```
Comment (skabashnyuk) [inline assembly/assembly-ide-war/pom.xml:435]: Do we need something like this https://github.com/codenvy/codenvy/blob/cbf975ecdff431b1feacd02cd6743fe2dec3e456/assembly/onpremises-ide-compiling-war-ide/pom.xml#L425 here

File: ide/che-core-ide-app/pom.xml [hunk 3]
```diff
@@ -26,13 +26,17 @@
         <dto-generator-out-directory>${project.build.directory}/generated-sources/dto/</dto-generator-out-directory>
         <findbugs.failonerror>false</findbugs.failonerror>
         <generated.sources.directory>${project.build.directory}/generated-sources/gen</generated.sources.directory>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (skabashnyuk) [inline ide/che-core-ide-app/pom.xml]: variable

**Review by skabashnyuk — COMMENTED** [review]
Ensure that jsr305 is not a part of any war or assembly

**Comment by svor** [comment]
> move gwt.compiler.version to separate property

we wouldn't use this property it isn't necessary

> Ensure that jsr305 is not a part of any war or assembly

excluded

> Do we need something like this https://github.com/codenvy/codenvy/blob/cbf975ecdff431b1feacd02cd6743fe2dec3e456/assembly/onpremises-ide-compiling-war-ide/pom.xml#L425 here

We need. I've added

**Review by vparfonov — APPROVED** [review]

**Review by dkulieshov — APPROVED** [review]
Splendid PR

**Review by dkulieshov — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Review by bmicklea — APPROVED** [review]

**Pull request merged**